### PR TITLE
[core] PARTY_MEMBERS_IN_ZONE: Ensure that PMember is not nullptr in Latent Effect Container (3)

### PR DESCRIPTION
**Pull request by CatsEyeXI**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Handles an edge case where calling `getZone()` may result in retrieving a nullptr.
image

## Steps to test these changes
Not entirely sure how to reproduce this reliably.

### CODE REVIEW

**Comment by WinterSolstice8** [comment]
There's a very brief window of time where loc/loc.zone is null when someone zones in. It realistically should never be null but I believe you when it's hard to produce

**Review by zach2good — APPROVED** [review]

**Comment by zach2good** [comment]
`loc` isn't a pointer type, so you can't compare it with `nullptr`:
https://github.com/LandSandBoat/server/blob/base/src/map/entities/baseentity.h#L309

If you're looking at a crash and it says `this` is nullptr, it means the current class, not where the crash is.

https://en.cppreference.com/w/cpp/language/this

You'll want to look one level up from here, where latents are being processed, where `GetZone` is being called on a `nullptr` PChar, or something like that

**Comment by zach2good** [comment]
This doesn't even compile because the code isn't valid?

https://godbolt.org/z/joeK5o38s

**Review by zach2good — CHANGES_REQUESTED** [review]
See inline comments

**Comment by zach2good** [comment]
Or even better, it's involved with `ReloadParty()`, RIP

**Comment by CatsEyeXI** [comment]
> See inline comments

I don't see any inline comments, but I think I've moved the nullptr checks to the right place in the latent effect container.

File: src/map/latent_effect_container.cpp [hunk 1]
```diff
@@ -455,14 +455,14 @@ void CLatentEffectContainer::CheckLatentsPartyMembers(size_t members, size_t tru
                     for (size_t m = 0; m < members; ++m)
                     {
                         auto* PMember = (CCharEntity*)m_POwner->PParty->members.at(m);
-                        if (PMember->getZone() == m_POwner->getZone())
+                        if (PMember != nullptr && PMember->getZone() == m_POwner->getZone())
```
Comment (zach2good) [inline src/map/latent_effect_container.cpp:458]: All of these are the same concept as your other PR: a `static_cast` will just stuff whatever type of pointer it finds into the shape you request, `dynamic_cast` will actually check that it's valid. If a non-player is here (and non-player parties exist, I think) this will force a Mob into the Char pointer.

Comment (zach2good) [reply src/map/latent_effect_container.cpp:458]: Inline comments: These, as part of the review.

Not inline comments: just comments on the PR

**Review by zach2good — CHANGES_REQUESTED** [review]

File: src/map/latent_effect_container.cpp (hunk 1 above)
Comment (CatsEyeXI) [reply src/map/latent_effect_container.cpp:458]: My apologies, can you please break it down a bit further for me, and tell me what exactly needs to change here?
Much appreciated.

Comment (zach2good) [reply src/map/latent_effect_container.cpp:458]: `auto* PMember = dynamic_cast<CCharEntity*>(m_POwner->PParty->members.at(m));`

Comment (CatsEyeXI) [reply src/map/latent_effect_container.cpp:458]: Thank you!

**Review by zach2good — APPROVED** [review]
